### PR TITLE
Improve --tests flag to support wildcards and module wise execution

### LINKED ISSUE #27258: [Testerina] Implement running tests of a specific module
**Description:**
Module wise testing is not supported in the test framework with the new project api changes

The following small adjustments need to be made 

- [ ] Include all Jars of the package when running test in each module
- [ ] Balo should only be created if the tests pass

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Improves the --tests flag. Adds Wildcard support and module wise execution.

Fixes #27258

## Samples

For a ballerina project `ProjectA` with module `Module1` each containing test cases.

```
ProjectA
├── main.bal
├── modules
│   └── Module1
│       ├── Module1.bal
│       └── tests
│           └── Module1_test.bal
└── tests
    └── main_test.bal
```

1. Run specific tests in the *Default* module only

```
$ ballerina test --tests <ProjectName>:<test>
```
example :

```
~ ballerina test --tests ProjectA:test1

~ ballerina test --tests ProjectA:A_test1,ProjectA:A_test2

// Run all tests starting with A_
~ ballerina test --tests ProjectA:A_*

// Run all tests in the default module
~ ballerina test --tests ProjectA:*

```

2. Run tests in a *specified module* only

```
$ ballerina test --tests <ProjectName>.<ModuleName>:<test>
```
example :

```
~ ballerina test --tests ProjectA.Module1:A_test1

~ ballerina test --tests ProjectA.Module1:m_test1,ProjectA.Module1:m_test2

// Run all tests starting with m_
~ ballerina test --tests ProjectA.Module1:m_*

// Run all tests in the specified module
~ ballerina test --tests ProjectA.Module1:*

```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java [hunk 1]
```diff
@@ -198,12 +198,38 @@ private static Test getTest(String testName, List<Test> tests) {
...
+    public static List<Test> getSingleExecutionTests(TestSuite suite, List<String> functions) {
+        List<String> filteredList = new ArrayList<>();
+
+        // Go through each function in the functionsList
+        for (String function : functions) {
+            if (function.contains(":")) {
+                String[] functionDetail = function.split(":");
+
+                if (functionDetail[0].equals(suite.getPackageID())) {
+                    if (functionDetail[1].equals(TesterinaConstants.WILDCARD)) {
```
Comment (NipunaRanasinghe) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java:214]: Don't we need to handle the potential `IndexOutOfBoundException`(s) in here?

Comment (Ibaqu) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java]: Yes. I had missed this. I have handled the exception and now it throws an error.

Comment (NipunaRanasinghe) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java]: Great!

**Comment by azinneera** [comment]
LGTM

**Review by azinneera — APPROVED** [review]

**Pull request merged**